examples/bicluster/plot_bicluster_newsgroups.py

```diff
@@ -41,18 +41,21 @@
 print(__doc__)
 
 
-def number_aware_tokenizer(doc):
-    """ Tokenizer that maps all numeric tokens to a placeholder.
+def number_normalizer(tokens):
+    """ Map all numeric tokens to a placeholder.
 
     For many applications, tokens that begin with a number are not directly
     useful, but the fact that such a token exists can be relevant.  By applying
     this form of dimensionality reduction, some methods may perform better.
     """
-    token_pattern = re.compile(u'(?u)\\b\\w\\w+\\b')
-    tokens = token_pattern.findall(doc)
-    tokens = ["#NUMBER" if token[0] in "0123456789_" else token
-              for token in tokens]
-    return tokens
+    for t in tokens:
+        yield re.sub(r"^\d+$", "#NUMBER", t)
```

Comment:
This is different behavior tham before. The docstring is no longer accurate: the old version checks if the **first character** is a digit, your regex requires the **entire token** to be made out of digits.

Either update the docstring or (maybe better) replace by

```
return ("#NUMBER" if token[0].isdigit() else token for token in tokens)
```

Come to think of it, even if you want to keep your behaviour, it is probably faster to use


```
return ("#NUMBER" if token.isdigit() else token for token in tokens)
```